### PR TITLE
Check more results returned by CrossRef API for matching

### LINKED ISSUE #2431: Can download article by DOI but not title
If I click the New Bibtex Entry button, paste this DOI:

    http://dx.doi.org/10.1016/j.engappai.2016.01.001

into the ID field, and then click the Generate button, then Jabref successfully downloads the article's BibTeX entry.

But if I instead paste the article's title into the ID field, change the ID Type to Title, and then click Generate, I get the error shown in this screenshot:

image

I'm using the title string:

    Traffic flow optimization: A reinforcement learning approach

This is with: 

    JabRef 3.8.1 windows 7 6.1 amd64  Java 1.8.0_111

### PR DESCRIPTION
**Pull request by stefan-kolb**
Resolves #2431

Changes:
- We will now check 5 results instead of just the first one for a matching title.

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/importer/fetcher/CrossRef.java [hunk 1]
```diff
@@ -84,33 +90,37 @@ private static String enhanceQuery(String query, BibEntry entry) {
...
+                // additional check, as sometimes subtitle is needed but sometimes only duplicates the title
+                if (data.getJSONArray("subtitle").length() > 0) {
+                    String dataWithSubTitle = dataTitle + " " + data.getJSONArray("subtitle").getString(0);
+
+                    if (editDistanceIgnoreCase(entryTitle, dataWithSubTitle) <= METRIC_THRESHOLD) {
+                        return Optional.of(data.getString("DOI"));
+                    }
+                }
+            } catch(JSONException ex) {
+                return Optional.empty();
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/importer/fetcher/CrossRef.java]: Mabye useful to add a logging here in case the json format changes?

Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/logic/importer/fetcher/CrossRef.java]: This should be detected by our Tests. Shouldn't it?

However, logging here does no harm.

Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/importer/fetcher/CrossRef.java]: The test will fail yes.

**Comment by matthiasgeiger** [comment]
LGTM

Do you want to implement https://github.com/JabRef/jabref/issues/2455 as well?  :heart_eyes_cat:

**Comment by matthiasgeiger** [comment]
Ok I interpret this as a "no" :wink:

**Comment by stefan-kolb** [comment]
nah will do that later, maybe at jabcon. is a separate task.

**Pull request merged**